src/formats/gif.zig

```diff
@@ -1017,4 +1102,434 @@ pub const GIF = struct {
+        };
+
+        @memcpy(&header.magic, MAGIC.ptr);
+        @memcpy(&header.version, VERSIONS[1].ptr);
+
+        if (image.width > std.math.maxInt(u16) or image.height > std.math.maxInt(u16)) {
+            return Image.WriteError.Unsupported;
+        }
+
+        header.width = @truncate(image.width);
+        header.height = @truncate(image.height);
+
+        var palette_buffer: [3 * 256]u8 = undefined;
+        var palette_byte_length: usize = 0;
+
+        const pixel_format = std.meta.activeTag(pixels.*);
+        if (pixel_format.isIndexed()) {
+            if (pixels.*.getPalette()) |palette_slice| {
+                if (palette_slice.len > 0) {
+                    const palette_index = try paletteSizeIndex(palette_slice.len);
+                    header.flags.use_global_color_table = true;
+                    header.flags.global_color_table_size = @intCast(palette_index);
+                    // Color resolution = bits per primary color - 1
+                    // Our palettes use RGBA32 (8 bits), so color_resolution = 7
+                    header.flags.color_resolution = 7;
+
+                    const entries = paletteEntryCounts[palette_index];
+                    palette_byte_length = entries * 3;
+                    encodeColorTable(palette_buffer[0..palette_byte_length], palette_slice, entries);
+                }
+            }
+        }
+
+        try writer.writeStruct(header, .little);
+
+        if (palette_byte_length > 0) {
+            try writer.writeAll(palette_buffer[0..palette_byte_length]);
+        }
+
+        // Write loop extension if loop_count is set (for animated GIFs or explicit requests)
+        // Always write if loop_count is AnimationLoopInfinite (-1) or > 0
+        // Note: loop_count == 0 means no extension (single play, no loop)
+        if (loop_count == Image.AnimationLoopInfinite) {
+            try writeLoopExtension(writer, 0); // GIF uses 0 to represent infinite looping
+        } else if (loop_count > 0) {
+            if (loop_count <= @as(i32, std.math.maxInt(u16))) {
+                try writeLoopExtension(writer, @intCast(loop_count));
+            }
+        }
+    }
 };
+
+const paletteEntryCounts = [_]usize{ 2, 4, 8, 16, 32, 64, 128, 256 };
+
+fn loopCountToExtension(loop_count: i32) Image.WriteError!?u16 {
+    if (loop_count == 0) {
+        // Value 0 means no extension should be written.
+        return null;
+    }
+    if (loop_count == Image.AnimationLoopInfinite) {
+        // GIF uses 0 to represent infinite looping.
+        return 0;
+    }
+    if (loop_count < 0) {
+        return Image.WriteError.Unsupported;
+    }
+    if (loop_count > @as(i32, std.math.maxInt(u16))) {
+        return Image.WriteError.Unsupported;
+    }
+    const converted: u16 = @intCast(loop_count);
+    return converted;
+}
+
+fn writeLoopExtension(writer: *std.Io.Writer, loop_count: u16) Image.WriteError!void {
+    try writer.writeAll(&[_]u8{
+        0x21, // Extension Introducer.
```

Comment:
You should either pack those into a extern struct for a data block header or use the enums already present in this file.

Example:
```zig
const data_block_header: ExtensionDataHeader = .{
.data_kind = .extension,
.extension_kind = .application_extension,
.block_size = 0x0b,
```